bundler/helpers/v1/monkey_patches/lazy_specification_patch.rb

```diff
@@ -0,0 +1,25 @@
+# typed: false
+# frozen_string_literal: true
+
+# Adds default_gem? method to Bundler::LazySpecification
+# https://github.com/rubygems/rubygems/blob/bundler-v2.5.9/lib/rubygems/basic_specification.rb#L96-L99
+
+require "bundler/lazy_specification"
+
+# Check if the gem is a default gem
+# Ensure loaded_from is not nil and check if the gem is located in the default specifications directory
+module LazySpecificationDefaultGemPatch
+  attr_accessor :loaded_from unless instance_methods.include?(:loaded_from)
```

Comment:
@bdragon : To fix the failing test in v1, I have replaced the `respond_to?` to with `attr_accessor` for safely adding the `loaded_from` method without overriding existing implementations if the method exists.